File: 14. Longest Common Prefix/14.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""
        
        mn = min(len(s) for s in strs)
        if mn == 0:
            return ""
        
        for k in range(mn):
            current_char = strs[0][k]
            for s in strs:
                if s[k] != current_char:
                    return strs[0][:k]
        
        return strs[0][:mn]

File: 14. Longest Common Prefix/14.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    if (!strs.length) {
        return ""
    }
    const mn = Math.min(...strs.map(s => s.length))

    for ( let k = 0; k < mn; k++){
        current_char = strs[0][k]
        for (const s = 0; s < strs.length; s++){
            if (s[k] != current_char){
                return strs[0].substring(0, k)
            }
        }
    }

    return strs[0].substring(0, mn)
};